### DISCUSSION TITLE
Get rid of type="module"

### DISCUSSION THREAD

**Discussion by AndrzejSala** [nrwl/nx#9062] [General]
Hi,
I would like to remove `type="module"` from output scripts attached to index.html and use them as normal commonJS scripts with `defer`.

![Screenshot 2022-02-21 at 16 57 09](https://user-images.githubusercontent.com/16742965/154989549-9fdf937a-241a-4868-b0a3-02dd801a3f5f.png)

So for example I would like to transform:
`<script src="main.760d655e0879b7ea.esm.js" type="module"></script>`
to:
`<script src="main.760d655e0879b7ea.esm.js" defer></script>`

How I can do that? Should I change nx, webpack or typescript config?

**Comment by hellokingdom** [comment]
@AndrzejSala Did you find solution to this problem?

**Comment by rfprod** [comment]
I doubt it is possible with the latest version of Angular without the use of crutches.
It's Angular CLI, see here:
- https://stackoverflow.com/questions/70880595/angular-13-adds-type-module-to-the-js-script-tags-in-index-html
- https://github.com/angular/angular-cli/issues/22159/

**Comment by mpoisot** [comment]
I'm facing a similar issue. I need to support very old browsers that don't load scripts with type=module. I'm on NX 13 using `@nrwl/web:webpack` executor to build a React app, so this is not an angular-specific problem.

I'm trying to figure out where exactly this transformation is happening. When running react dev server, script tags injected at the end of <body> (runtime.js, main.js, polyfill.js, etc) use `defer`. When I run nx build to create a production build, those <script> tags use `type="module"` instead.

Any help appreciated.

**Reply by mpoisot** [reply]
I ended up fixing this by using HtmlWebpackPlugin in my custom webpack config, and disabling Nx's default index.html generation by adding `"generateIndexHtml": false` to build options in project.json.
